src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
Good, that was my conclusion too. If that's the case for 1), then what do you think about this?
- Attach the node's id to the `dht_immutable_item` structure when inserted, and that way I can perform a "virtual" pseudo-partitioning while looking for the last one to remove it. (I just mean to trick the comparator that for all the items stored with an id different from the current one in consideration, returns `-1`)

The downside is the extra memory.